src/mo_types/cons.ml

```diff
@@ -41,11 +51,38 @@ let unsafe_set_kind c k = c.kind := k
 let name c = c.name
 
 let to_string show_stamps sep c =
```

Comment:
I would revert to_string and may add a flag to show the scopes for debugging only. Then most of the test would change bug still be readable.

Reply:
Unfortunately, reverting `to_string` will cause errors such as non-productive types or duplicate type definitions, hence we did it this way. From our observations, the output of `to_string` must also be parseable as a variable, or some tests will fail, hence we substitute some characters.